### DISCUSSION TITLE
Failover scenario for npgsql

### DISCUSSION THREAD

**Discussion by j0shuani** [linq2db/linq2db#4474] [Q&A] [answered]
Starting from version npgsql 7.0, setting up multiple hosts for failover occurs through the `NpgsqlDataSourceBuilder + BuildMultiHost` command. Now I looked at the linq2db source code and it still uses the creation of a connection through the `NpgsqlConnection` class.

Has anyone checked whether failover works on the current version of linq2db?

Thank you in advance.

**Comment by MaceWindu** [comment, marked as answer]
You can use it with `UseConnectionFactory` setting

```cs
new DataOptions().UseConnectionFactory(o => dataSource.CreateConnection())
```

**Comment by j0shuani** [comment]
Hi @MaceWindu,

Thanks a lot.
